Next 14 Support

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @logtail/next@0.1.0
npm ERR! Found: next@14.0.2
npm ERR! node_modules/next
npm ERR!   next@"^14.0.2" from the root project
npm ERR!   peer next@"*" from next-themes@0.2.1
npm ERR!   node_modules/next-themes
npm ERR!     next-themes@"^0.2.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer next@"^12.1.4 || ^13" from @logtail/next@0.1.0
npm ERR! node_modules/@logtail/next
npm ERR!   @logtail/next@"^0.1.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: next@13.5.6
npm ERR! node_modules/next
npm ERR!   peer next@"^12.1.4 || ^13" from @logtail/next@0.1.0
npm ERR!   node_modules/@logtail/next
npm ERR!     @logtail/next@"^0.1.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! /Users/alexmarmon/.npm/_logs/2023-11-11T18_38_19_451Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alexmarmon/.npm/_logs/2023-11-11T18_38_19_451Z-debug-0.log
```

Forcing the install works as a workaround but full support would be awesome!

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,6 +1,7 @@
 name: CI
 
 on:
+  workflow_dispatch:
   pull_request:
     branches:
       - main

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,21 +21,11 @@ jobs:
           node-version: 20
       - run: npm install
       - run: npm run check-format
-  unittests:
-    name: Check Tests
+  build-and-test:
+    name: Build and Test
     runs-on: ubuntu-latest
     env:
       LOGTAIL_SOURCE_TOKEN: https://example.co/api/test
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 20
-      - run: npm install
-      - run: npm test
-  build:
-    name: Build
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         node:
@@ -52,11 +42,12 @@ jobs:
           node-version: ${{ matrix.node }}
       - run: npm install
       - run: npm run build
+      - run: npm test
   publish:
     name: Publish
     needs: 
       - check-format
-      - build
+      - build-and-test
     runs-on: ubuntu-latest
     if: startsWith(github.ref, 'refs/tags/')
     steps:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -36,12 +36,15 @@ jobs:
           - 18.x
           - 19.x
           - 20.x
+        install-cmd:
+          - npm install
+          - npm update
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node }}
-      - run: npm install
+      - run: ${{ matrix.install-cmd }}
       - run: npm run build
       - run: npm test
   publish:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -18,7 +18,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 17
+          node-version: 20
       - run: npm install
       - run: npm run check-format
   unittests:
@@ -30,7 +30,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: 17
+          node-version: 20
       - run: npm install
       - run: npm test
   build:
@@ -42,6 +42,9 @@ jobs:
           - 15.x
           - 16.x
           - 17.x
+          - 18.x
+          - 19.x
+          - 20.x
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3

--- a/package.json
+++ b/package.json
@@ -32,7 +32,7 @@
   },
   "homepage": "https://github.com/logtail/logtail-nextjs#readme",
   "peerDependencies": {
-    "next": "^12.1.4 || ^13"
+    "next": "^12.1.4 || ^13 || ^14"
   },
   "devDependencies": {
     "@types/jest": "^27.5.1",

